contracts/vaults/Pool.sol

```diff
@@ -65,7 +65,7 @@ contract Pool is Base {
             }));
         }
         poolInfos.push(PoolInfo({
-            lpToken: IERC20(_lpToken),
+            lpToken: IERC20Upgradeable(_lpToken),
```

Comment:
cam just be IERC20

Reply:
It gives an error deploying:
Error: Contract `HATVaults` is not upgrade safe

@openzeppelin/contracts/utils/Address.sol:191: Use of delegatecall is not allowed

Also even if can ignore the error I think it's best practice to use only the upgradeable repo with upgradeable contracts

Reply:
The error must be a bug, right?  In any case, I appreciete your point about using just a single dependency